### DISCUSSION TITLE
How to configure the behavior of a multi-key sequence? Can it be done using a custom kitten?

### DISCUSSION THREAD

**Discussion by thenbe** [kovidgoyal/kitty#6857] [Q&A] [answered]
I often found myself shying away from customizing and depending on kitty's mappable actions because the keymaps often end up conflicting with nvim, OS, or zsh's keybindings. Now, I'm trying to figure out a way to have them co-exist peacefully.

For example, I'd like to use `ctrl+space` as a "leader" key to several commands, but only when I'm not in nvim. In nvim, I'd like to effectively "ignore" the following maps since (1) I have `ctrl+space` mapped to something else, (2) I don't wait to wait for the slight pause, and (3) I have no use for some of these commands in my editor anyways.

```
map ctrl+space>f kitten hints --type path --program -
map ctrl+space>shift+f kitten hints --type path
map ctrl+space>l kitten hints --type line --program -
map ctrl+space>w kitten hints --type word --program -
map ctrl+space>h kitten hints --type hash --program -
```

In short, is there way to have the previous maps "active" only when not in nvim?

I did manage to figure out a solution for simple maps by using a custom kitten (largely based on [`pass_keys.py`](https://github.com/knubie/vim-kitty-navigator/blob/master/pass_keys.py)). But when it comes to multi-key sequences, I'm not sure what a "clean" approach looks like (or if one even exists).

My attempts with custom kittens and `send_text` were not successful; I can't seem to find a way to avoid the pause when in nvim. Ideally, there'd be a way to "intercept" the sequence during the pause, insert some simple logic to check the window, then decide whether to skip the pause or keep waiting for a potential second leg of a multi-key sequence.

**Comment by kovidgoyal** [comment]
Not sure I understand your issue. If I were implementing this, I would
do so in a watcher. React to the focus change and user variable events.
Have neovim set a user vaiarble when it start and unset it when it quits
using the escape code. In your watcher when that user variable is set
unload the current mappings and load your neovim specific mappings.

I dont see what difference mappings being singe or multiple key needs to
make and I am not sure why you need send_text at all.

**Reply by kovidgoyal** [reply, marked as answer]
And just for the record in master kitty supports defining keyboard shortcuts based on the currently active window/program out of the box. See mappings.rst for documentation.
